Update with pub constraint change (#2727)
dart-lang/site-www: src/tools/pub/pubspec.md

Pub tries to find the latest version of a package whose SDK constraint works
with the version of the Dart SDK that you have installed.

> As of Dart 2.12, it is an error to have a pubspec without an SDK constraint,
> and when omitted `pub get` will fail with an error:

Reviewer: ```suggestion
As of Dart 2.12, omitting the SDK constraint is an error.
When the pubspec has no SDK constraint,
`pub get` fails with a message like the following:
```
